packages/cli-repl/src/cli-repl.ts

```diff
@@ -149,6 +157,15 @@ class CliRepl {
       this.warnAboutInaccessibleFile(err);
     }
 
+    if (driverOptions.autoEncryption) {
+      const extraOptions = {
+        ...(driverOptions.autoEncryption.extraOptions ?? {}),
+        ...(await this.startMongocryptd())
+      };
+
+      driverOptions.autoEncryption = { ...driverOptions.autoEncryption, extraOptions };
+    }
```

Comment:
This mirrors what we do in the `Mongo` class, and was missing before (i.e. it would always only try the system `mongocryptd` before).

I feel like this is a symptom of a bigger problem, and that – if possible – we should be using the `Mongo` constructor after initializing the REPL/before starting it to create the connection (as opposed to having the same code in two places).

Reply:
Makes sense to me, would you recommend doing this here / create a ticket?

Reply:
Sure, done: https://jira.mongodb.org/browse/MONGOSH-661